src/app/shared/components/workshop-card/workshop-card.component.ts

```diff
@@ -108,6 +110,25 @@ export class WorkshopCardComponent implements OnInit, OnDestroy {
     this.isFavorite = !this.isFavorite;
   }
 
+  onChange(): void {
+    const dialogRef = this.dialog.open(ConfirmationModalWindowComponent, {
+      width: Constants.MODAL_SMALL,
+      data: {
+        type: this.workshopData.status === 'Open' ? ModalConfirmationType.closeSet : ModalConfirmationType.openSet,
+      },
+    });
+    dialogRef.afterClosed().subscribe((result: boolean)  => {
+      if(result){
+        if(this.workshopData.status === 'Open') {
+          this.workshopData.status = 'Closed';
+        }else if (this.workshopData.status === 'Closed'){
```

Comment:
it is better to create a const Open/Close and assign it to the object afterwords